### PR TITLE
Improved upload UI

### LINKED ISSUE #226: Improve Video Upload UX
This issue combines #63, #183 and #184 

### acceptance criteria:

- [ ] Add "Add Videos from Dropbox" button to collection detail page (https://projects.invisionapp.com/boards/SG3980TYRDCQM/)
- [ ] replace progress bar from upload dialog with video status
  - [ ] for status of `CREATED` or `UPLOADING` or `TRANSCODING`, display "In progress. We will e-mail you when your video is ready"
  - [ ] for status `COMPLETE` display the video thumbnail
  - [ ] for any other status, display "Something went wrong. We are working on it."
- [ ] Don’t let the video title float up to the top of the thumbnail box when there’s no thumbnail
- [ ] don't wrap the buttons [when the collection description gets too long](https://user-images.githubusercontent.com/430126/30346831-1185ba96-97d8-11e7-977c-a38b65d42068.png)

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?
Closes #226

#### What's this PR do?
Improves the upload UI to be more seamlessly integrated into the rest of the app

#### How should this be manually tested?
Upload a file via the Dropbox button, make sure the the page reloads after your file has been chosen and submitted

#### Where should the reviewer start?
`CollectionDetailPage.js` and `VideoCard.js`

#### Any background context you want to provide?
- I added [react-dropbox-chooser](https://github.com/sdoomz/react-dropbox-chooser), a very small NPM package, to this project. Unfortunately it uses `React.PropTypes`, which is deprecated and triggers a warning. If this becomes a deal-breaker, we can very easily fork the package and fix the ~8 offending lines.
- It seems like we might want to avoid putting the dropbox key value into the front-end `SETTINGS` variable in some cases, but I'm not exactly sure what those cases are (Users that don't have admin permission for the specific collection being viewed? Users that have no admin permissions for any collections? etc.). For now it gets set for all users.
- I'll wait for word from @roberthouse54 or @pdpinch about the preferred text for error/loading messages.
- I'm still unable to successfully upload locally, hence the screenshot.

#### Screenshots (if appropriate)
ss 2017-09-15 at 11 30 52

### CODE REVIEW

File: static/js/containers/CollectionDetailPage.js [hunk 1]
```diff
@@ -112,10 +123,22 @@ class CollectionDetailPage extends React.Component {
...
         <div className="tools">
           {
-            collection.is_admin &&
-            <a onClick={ showDialog.bind(this, DIALOGS.NEW_COLLECTION) }>
-              <i className="material-icons">settings</i>
-            </a>
+            collection.is_admin && [
+              <DropboxChooser
+                key="upload"
+                appKey={SETTINGS.dropbox_key}
```
Comment (pdpinch) [inline static/js/containers/CollectionDetailPage.js]: Only users with admin permission are allowed to upload a file from dropbox, so we could limit this key to members of the admin group. 

However, I don't think there's much risk associated with leaking this key. It's our app key right? That's just how dropbox identifies us in order to manage our access to their APIs. I think the worst case of leaking it is that someone else uses it and DOS's us by going over quota. It doesn't access any of our resources.

File: static/js/components/VideoCard.js [hunk 2]
```diff
@@ -13,24 +14,53 @@ type VideoCardProps = {
...
+
+  if (isProcessing) {
+    videoDisplay = <div>
+      <p>Upload in progress.</p>
+      <p>You should receive an email when the upload is completed.</p>
+    </div>;
+  } else if (hasError) {
+    videoDisplay = <div className="error">
+      <p>Upload failed.</p>
+      <p>Please contact the ODL Video Services team.</p>
```
Comment (pdpinch) [inline static/js/components/VideoCard.js]: We should include a mailto link to the address in `ODL_VIDEO_SUPPORT_EMAIL`

File: static/js/containers/CollectionDetailPage.js [hunk 3]
```diff
@@ -112,10 +123,22 @@ class CollectionDetailPage extends React.Component {
...
+                success={this.handleUpload}
+                multiselect={true}
```
Comment (mbertrand) [inline static/js/containers/CollectionDetailPage.js]: Try adding this option here:  `extensions: ['video']`

This was on the old HTML template form and only allowed video files to be chosen.  With the new chooser I'm allowed to upload any file.

File: static/js/containers/CollectionDetailPage.js [hunk 4]
```diff
@@ -112,10 +123,22 @@ class CollectionDetailPage extends React.Component {
         </div>
         <div className="tools">
           {
-            collection.is_admin &&
-            <a onClick={ showDialog.bind(this, DIALOGS.NEW_COLLECTION) }>
-              <i className="material-icons">settings</i>
-            </a>
+            collection.is_admin && [
+              <DropboxChooser
```
Comment (mbertrand) [inline static/js/containers/CollectionDetailPage.js]: This chooser is getting incorrect urls for selected videos.  For example, selecting a particular video on master branch gets a source_url of `https://dl.dropboxusercontent.com/1/view/r07a4tunjwlqc97/jellie2s.mp4` but on this branch gets `https://www.dropbox.com/s/g5s28d1723b8kvr/jellie2s.mp4?dl=0`, which is an HTML page with the video file embedded in it. However for some reason the video status is not being changed to 'Upload failed' (I'll open another issue for that soon).

Comment (mbertrand) [reply static/js/containers/CollectionDetailPage.js]: PS This option will fix it: `linkType = 'direct'`

File: static/js/containers/CollectionDetailPage.js (hunk 1 above)
Comment (gsidebo) [reply static/js/containers/CollectionDetailPage.js]: The issue is that a user can be an admin for a specific set of collections, but that app key is obviously constant for all collections. We _could_ just tack on the dropbox key to any request for a specific collection where the user happens to be an admin, but that's a hack and could introduce some complications in the front end logic. Given that this is app requires a login and there's low risk even if the app key is leaked, I'm fine just leaving it as-is

File: static/js/components/VideoCard_test.js [hunk 5]
```diff
@@ -5,6 +5,12 @@ import { shallow } from 'enzyme';
 import { assert } from 'chai';
 
 import VideoCard from './VideoCard';
+import { makeVideoThumbnailUrl } from "../lib/urls";
+import {
+  VIDEO_STATUS_UPLOADING,
+  VIDEO_STATUS_UPLOAD_FAILED,
```
[the remark this answers is not in the document]
Comment (gsidebo) [reply static/js/components/VideoCard_test.js]: I went one better and stubbed the helper functions that return the given statuses

**Comment by mbertrand** [comment]
@gsidebo looks great, Ferdi just requested that the progress text for videos in process be changed to:

> **In progress...**
> You should receive an e-mail when the video is ready.

Once that change is made go ahead and merge.

**Pull request merged**